src/openrct2/windows/ride.c

```diff
@@ -3794,14 +3794,16 @@ static void window_ride_maintenance_dropdown(rct_window *w, rct_widgetindex widg
                     break;
                 for (sint32 i = 0; i < ride->num_vehicles; ++i) {
                     uint16 spriteId = ride->vehicles[i];
-                    do {
-                        vehicle = GET_VEHICLE(spriteId);
-                        vehicle->update_flags &= ~(
-                            VEHICLE_UPDATE_FLAG_BROKEN_CAR |
-                            VEHICLE_UPDATE_FLAG_7 |
-                            VEHICLE_UPDATE_FLAG_BROKEN_TRAIN
-                            );
-                    } while ((spriteId = vehicle->next_vehicle_on_train) != 0xFFFF);
+                    if (spriteId != SPRITE_INDEX_NULL) {
+                        do {
+                            vehicle = GET_VEHICLE(spriteId);
+                            vehicle->update_flags &= ~(
+                                VEHICLE_UPDATE_FLAG_BROKEN_CAR |
+                                VEHICLE_UPDATE_FLAG_7 |
+                                VEHICLE_UPDATE_FLAG_BROKEN_TRAIN
+                                );
+                        } while ((spriteId = vehicle->next_vehicle_on_train) != 0xFFFF);
```

Comment:
This already checks for `SPRITE_INDEX_NULL`, is there no better way to format this loop?